Source/Application/Kysect.Shreks.Application.Handlers/Github/CreateOrUpdateGithubSubmissionHandler.cs

```diff
@@ -51,17 +61,28 @@ public async Task<Response> Handle(Command request, CancellationToken cancellati
 
         if (submission is null || submission.IsRated)
         {
+            if (triggeredByAnotherUser && !triggeredByMentor)
+            {
+                var message = $"Repository {repository} is assigned to another student. Close this pull request";
+                throw new DomainInvalidOperationException(message);
+            }
             submission = await CreateSubmission(request, cancellationToken);
             isCreated = true;
             _tableUpdateQueue.EnqueueSubmissionsQueueUpdate(submission.GetCourseId(), submission.GetGroupId());
         }
-        else if (!await TriggeredByMentor(userId, request.PullRequestDescriptor.Organization))
+        else if (!triggeredByMentor)
         {
             submission.SubmissionDate = Calendar.CurrentDate;
 
             _context.Submissions.Update(submission);
             await _context.SaveChangesAsync(cancellationToken);
             _tableUpdateQueue.EnqueueSubmissionsQueueUpdate(submission.GetCourseId(), submission.GetGroupId());
+
+            if (triggeredByAnotherUser)
+            {
+                var message = $"Repository {repository} is assigned to another student";
```

Comment:
+ "Do not user {sender} account for this repository. Submission date will be update."